pkg/config/configuration.go

```diff
@@ -242,7 +242,8 @@ type OtelSpec struct {
 	Protocol        string `json:"protocol,omitempty" yaml:"protocol,omitempty"`
 	EndpointAddress string `json:"endpointAddress,omitempty" yaml:"endpointAddress,omitempty"`
 	// Defaults to true
-	IsSecure *bool `json:"isSecure,omitempty" yaml:"isSecure,omitempty"`
+	IsSecure *bool             `json:"isSecure,omitempty" yaml:"isSecure,omitempty"`
+	Headers  map[string]string `json:"headers,omitempty" yaml:"headers,omitempty"`
```

Comment:
The name `headers` follows the same from OTel collector

https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/configgrpc/README.md#client-configuration